logic/LogicManager.java

```diff
@@ -104,21 +104,20 @@ private FunctionalIterator<Rule> rulesWithNegations() {
     }
 
     /**
-     * On commit we must clear the rule cache and revalidate rules - this will force re-running type resolution
+     * On commit we must clear the rule cache and revalidate rules - this will force re-running type inference
      * when we re-load the Rule objects
      * Rule indexes should also be deleted and regenerated as needed
      * Note: does not need to be synchronized as only called by one schema transaction at a time
      */
     public void revalidateAndReindexRules() {
         logicCache.rule().clear();
 
-        // re-validate all rules are valid
-        rules().forEachRemaining(rule -> rule.validate(this, conceptMgr));
+        if (graphMgr.schema().typesModified()) {
+            // re-validate all rules are valid
+            rules().forEachRemaining(rule -> rule.validate(this, conceptMgr));
 
-        // re-index if rules are valid and satisfiable
-        if (graphMgr.schema().rules().conclusions().isOutdated()) {
-            graphMgr.schema().rules().all().forEachRemaining(s -> fromStructure(s).reindex());
-            graphMgr.schema().rules().conclusions().outdated(false);
+            // recreate rule index conclusions
+            graphMgr.schema().rules().all().forEachRemaining(s -> fromStructure(s).conclusion().reindex());
```

Comment:
we only need to validate & reindex rules if the schema types are modified. The other validation (eg. negation cycle) detection must still be done